### PR TITLE
import and formula fixes

### PR DESCRIPTION
**Pull request by xithan**

### CODE REVIEW

File: mod/metrics/set/right/variables.rb [hunk 1]
```diff
@@ -74,13 +91,14 @@ def variable_row item_name, index, args
...
+  view :missing do
+    return super() unless card.new_card?
+    if card.formula_card
       card.extract_metrics_from_formula
-      render!(@denied_view, args)
+      render! @denied_view
     else
-      super(args)
+      Auth.as_bot { card.save! }
+      render! :core
```
Comment (ethn) [inline mod/metrics/set/right/variables.rb]: probably should be `@denied_view`, not `:core`, no?

File: mod/metrics/set/right/variables.rb [hunk 2]
```diff
@@ -48,7 +46,26 @@ def input_metric_name_by_index index
...
+        button_tag class: "pointer-item-add slotter",
+                   situation: "outline-secondary",
+                   data: { toggle: "modal", target: target },
+                   href: path(layout: "modal", view: :edit,
+                              mark: card.name,
+                              slot: { title: "Choose Metric" }) do
+          fa_icon(:plus) + " add metric"
+        end
+    end
+  end
```
Comment (ethn) [inline mod/metrics/set/right/variables.rb]: I like this move.

File: lib/capistrano/tasks/delayed_job.cap [hunk 3]
```diff
@@ -0,0 +1,25 @@
+namespace :delayed_job do
+  desc "stop delayed job processes"
+  task :stop do
+    on roles(:app) do
+      within release_path do
+        execute "pwd"
+        env = "env RAILS_ENV=production"
+        num_jobs = fetch :num_delayed_job_processes
+        execute :rvm, "default do", env, "./bin/delayed_job -n", num_jobs, "stop"
```
Comment (ethn) [inline lib/capistrano/tasks/delayed_job.cap:9]: I don't love how I did this, but it wasn't working if I called it without the rvm nonsense.

Comment (ethn) [reply lib/capistrano/tasks/delayed_job.cap:9]: if methods work in .cap, this would be a nice use case!

File: mod/metrics/set/right/variables.rb [hunk 4]
```diff
@@ -36,10 +36,8 @@ def input_metric_name_by_index index
   item_cards.fetch(index, nil).name
 end
 
-
-
 format :html do
-  view :core do |args|
+  view :core, chache: :never do |args|
```
Comment (ethn) [inline mod/metrics/set/right/variables.rb]: typo

Comment (ethn) [reply mod/metrics/set/right/variables.rb]: why wouldn't caching work here?

**Review by ethn — CHANGES_REQUESTED** [review]
some small changes requested.

This has all the code that you added to production today, yes?

File: mod/metrics/set/right/variables.rb (hunk 4 above)
Comment (xithan) [reply mod/metrics/set/right/variables.rb]: I wanted to make sure that that wasn't the reason that updates on this card had no effect

Comment (ethn) [reply mod/metrics/set/right/variables.rb]: I'm going to guess `chache: :never` didn't fix it :)

Comment (ethn) [reply mod/metrics/set/right/variables.rb]: FYI, view caching is off in development mode by default, so if it's breaking in development (and you haven't explicitly turned it on) it's probably not view caching's fault

**Comment by xithan** [comment]
I'll try again to get semaphore to work

**Pull request closed without merging**